lib/ace/ext/papeeria/biblio_completer.coffee

```diff
@@ -39,76 +37,71 @@ define((require, exports, module) ->
       @openPopup(@editor, prefix, keepPopupPosition)
 
     insertMatch: (editor, data) =>
-      if not data?
-        data = @popup.getData(@popup.getRow())
-      if not data?
+      data ?= @popup.getData(@popup.getRow())
+      if not data? or not data.action?
         @detach()
         return
-
-      if data.name == "Create new file"
-        @_doCreate()
-      else
-        @_doInsert(data.file)
+      data.action()
 
     detach: =>
       @editor.completer = new Autocomplete.Autocomplete()
       super()
 
 
   class BiblioCompleter
-    constructor: () ->
-      @enabledMendeley = false
-
-    setEnabledMendeley: (enabledMendeley) => @enabledMendeley = enabledMendeley
+    constructor: (@enabledMendeley, @searchLabel, @importLabel, @importSource, @searchCallback) ->
+      @isDisposable = true
 
     getCompletions: (editor, session, pos, prefix, callback) =>
       token = session.getTokenAt(pos.row, pos.column)
 
       if LatexParsingContext.isType(token, "cite")
-        default_result = [{
-          name: "Search"
-          snippet: "Search " + prefix
+        defaultResult = [{
+          name: @searchLabel
+          snippet: @searchLabel + prefix
           meta: ""
           score: 1000
           meta_score: 10
           completer: this
+          action: (editor) => @_doSearch(editor)
         }]
         # We don't show Import option to free users
-        if @enabledMendeley
-          default_result.push({
-            name: "Import"
-            snippet: "Import " + prefix
-            meta: "from Mendeley"
+        if @enabledMendeley?
+          defaultResult.push({
+            name: @importLabel
+            snippet: @importLabel + prefix
+            meta: @importSource
             score: 1000
```

Comment:
you sure a thousand is enough? I used `Number.MAX_VALUE`. 
and please extract it to a constant :)
also make sure it's okay that both import and search objects have the same `score`